src/ynEIGEN/ynEigen.sol

```diff
@@ -0,0 +1,320 @@
+
+    //--------------------------------------------------------------------------------------
+    //----------------------------------  VARIABLES  ---------------------------------------
+    //--------------------------------------------------------------------------------------
+
+    mapping(address => Asset) public assets;
+
+    IEigenStrategyManager eigenStrategyManager;
+    IAssetRegistry assetRegistry;
+
+    bool public depositsPaused;
+
+    //--------------------------------------------------------------------------------------
+    //----------------------------------  INITIALIZATION  ----------------------------------
+    //--------------------------------------------------------------------------------------
+
+    constructor() {
+       _disableInitializers();
+    }
+
+    struct Init {
+        string name;
+        string symbol;
+        IAssetRegistry assetRegistry;
+        IEigenStrategyManager eigenStrategyManager;
+        address admin;
+        address pauser;
+        address unpauser;
+        address[] pauseWhitelist;
+    }
+
+    function initialize(Init calldata init)
+        public
+        notZeroAddress(address(init.assetRegistry))
+        notZeroAddress(address(init.admin))
+        initializer {
+        __AccessControl_init();
+        __ReentrancyGuard_init();
+        __ynBase_init(init.name, init.symbol);
+
+        _grantRole(DEFAULT_ADMIN_ROLE, init.admin);
+        _grantRole(PAUSER_ROLE, init.pauser);
+        _grantRole(UNPAUSER_ROLE, init.unpauser);
+
+        assetRegistry = init.assetRegistry;
+        eigenStrategyManager = init.eigenStrategyManager;
+
+        _setTransfersPaused(true);  // transfers are initially paused
+        _updatePauseWhitelist(init.pauseWhitelist, true);
+    }
+
+
+    //--------------------------------------------------------------------------------------
+    //----------------------------------  DEPOSITS   ---------------------------------------
+    //--------------------------------------------------------------------------------------
+
+    /**
+     * @notice Deposits a specified amount of an asset into the contract and mints shares to the receiver.
+     * @dev This function first checks if the asset is supported, then converts the asset amount to unitOfAccount equivalent,
+     * calculates the shares to be minted based on the unitOfAccount value, mints the shares to the receiver, and finally
+     * transfers the asset from the sender to the contract. Emits a Deposit event upon success.
+     * @param asset The ERC20 asset to be deposited.
+     * @param amount The amount of the asset to be deposited.
+     * @param receiver The address to receive the minted shares.
+     * @return shares The amount of shares minted to the receiver.
+     */
+    function deposit(
+        IERC20 asset,
+        uint256 amount,
+        address receiver
+    ) public nonReentrant returns (uint256 shares) {
```

Comment:
The deposit function is vulnerable to share inflation attacks. Adding a minShares parameter to the deposit function can prevent sandwich attacks that could significantly reduce the number of shares minted to a user.

Reply:
corrected in base branch